src/autocomplete/WildcardMentionItem.js

```diff
@@ -0,0 +1,166 @@
+/* @flow strict-local */
+import React, { useCallback, useContext, useMemo } from 'react';
+import type { Node } from 'react';
+import { View } from 'react-native';
+import * as typeahead from '@zulip/shared/js/typeahead';
+
+import type { GetText, Narrow } from '../types';
+import { IconWildcardMention } from '../common/Icons';
+import ZulipText from '../common/ZulipText';
+import Touchable from '../common/Touchable';
+import { createStyleSheet, ThemeContext } from '../styles';
+import { caseNarrowDefault } from '../utils/narrow';
+import { TranslationContext } from '../boot/TranslationProvider';
+
+/**
+ * A type of wildcard mention recognized by the server.
+ *
+ * For the string the server knows this by, see serverCanonicalStringOf.
+ *
+ * See user doc on wildcard mentions:
+ *   https://zulip.com/help/pm-mention-alert-notifications#wildcard-mentions
+ */
+export enum WildcardMentionType {
+  // These values are arbitrary. For the string the server knows these by,
+  // see serverCanonicalStringOf.
+  All = 0,
+  Everyone = 1,
+}
+
+/**
+ * The canonical English string, like "everyone" or "all".
+ *
+ * See also serverCanonicalStringOf for the string the server knows this by.
+ * (It might differ in a future where servers localize these.)
+ */
+// All of these should appear in messages_en.json, so we can make the
+// wildcard mentions discoverable in the people autocomplete in the client's
+// own language. See getFilteredSuggestions.
+const englishCanonicalStringOf = (type: WildcardMentionType): string => {
+  switch (type) {
+    case WildcardMentionType.All:
+      return 'all';
+    case WildcardMentionType.Everyone:
+      return 'everyone';
+  }
+};
+
+/**
+ * The string recognized by the server, like "everyone" or "all".
+ *
+ * Currently the same as englishCanonicalStringOf, but that should change if
+ * servers start localizing these.
+ */
+export const serverCanonicalStringOf = englishCanonicalStringOf;
+
+export const descriptionOf = (
+  type: WildcardMentionType,
+  destinationNarrow: Narrow,
+  _: GetText,
+): string => {
+  switch (type) {
+    case WildcardMentionType.All:
+    case WildcardMentionType.Everyone:
+      return caseNarrowDefault(
+        destinationNarrow,
+        { topic: () => _('Notify stream'), pm: () => _('Notify recipients') },
+        // A "destination narrow" should really be a conversation narrow
+        // (i.e., stream or topic), but including this default case is easy,
+        // and better than an error/crash in case we've missed that somehow.
+        () => _('Notify everyone'),
+      );
+  }
+};
+
+export const getFilteredSuggestions = (
+  query: string,
+  destinationNarrow: Narrow,
+  _: GetText,
+): $ReadOnlyArray<WildcardMentionType> =>
+  Array.from(WildcardMentionType.members()).filter(type =>
```

Comment:
Sure, I can have it take the first match in [all, everyone, stream]. Is that a good order, or should we take the first match in [stream, all, everyone] or something else? (We'd exclude "stream" when the destination narrow isn't a stream or topic narrow, of course.)

If I change the code so it just shows one match, I see an odd thing where "all" shows up as the only wildcard-mention suggestion, when my actual query was "stream". That's because "all" matches by its `descriptionOf(type, …)`, which is `_('Notify stream')`. We should probably prevent that by removing the description search.

Reply:
Sounds good to me.

The help center mentions @-all, sometimes also @-everyone, and never @-stream that I can see:
https://zulip.com/help/mention-a-user-or-group
https://zulip.com/help/pm-mention-alert-notifications
So I think the right order of preference is [all, everyone, stream].